### ISSUE TITLE
Support for Docker compose files

### ISSUE THREAD

**Issue by greysteil** [dependabot/dependabot-core#390]
_From @armin-joellenbeck on December 23, 2017 9:21_

Knowing when new Docker images are published would be helpful when the are used in a Docker compose file too.

Just like #20, with the file `docker-compose.yml` instead of `Dockerfile`.

_Copied from original issue: dependabot/feedback#66_

**Comment by greysteil** [comment]
👍  for this, and should be relatively straightforward. A couple of things I want to get to first, but I'm definitely game for adding this to Dependabot at some point!

**Comment by victorcerutti** [comment]
Hi,
This would be a really good addition, is this still going to happen someday ?

Thanks in advance

**Comment by greysteil** [comment]
I hope so, yes! This is another one that @hmarr owns on our side, but he's very busy scaling Dependabot up to 100m repos!

**Comment by bkaid** [comment]
@stalebot please leave this open - hoping it gets implemented

**Comment by stefanb** [comment]
Instead of just hardcoding the `docker-compose.yml` filename it should be possible to specify a file mask or explicit list of files to check within the project.

**Comment by jimmyko** [comment]
I want this feature too

**Comment by valentijnscholten** [comment]
Are there any plans to finish the feature similar to what Renovate has?

**Comment by weapp** [comment]
There are any plan to add this?

**Comment by jurre** [comment]
> There are any plan to add this?

We don't have any plans to add support in the near future. Eventually we'd love to support it, but currently our team doesn't have any capacity to work on this or even properly review and maintain a community contribution for this

**Comment by jokay** [comment]
This issue was created almost 3 years ago and there seems no progress so far, but [Renovate Bot](https://docs.renovatebot.com/modules/manager/docker-compose) may be a solution until it's finally available in dependabot.

Hopefully we may see support for **docker-compose.yml** files one day :monocle_face:

**Comment by ulgens** [comment]
@jurre Any luck so far? 🙂

**Comment by jokay** [comment]
> @jurre Any luck so far? slightly_smiling_face

Btw [Renovate](https://github.com/renovatebot/renovate) works perfectly for `docker-compose.yml` too :wink:

---
![image](https://user-images.githubusercontent.com/18613935/143767568-559233df-348f-43f6-aa73-1b6fbc74bc5c.png)
---
![image](https://user-images.githubusercontent.com/18613935/143767602-28f73da3-7dd8-49af-a09e-fc27650ea01a.png)

**Comment by banesullivan** [comment]
Renovate seems like a seriously powerful Dependabot alternative, though I wasn't thrilled with its self-hosted/GitHub Actions support, I wanted to see how to do this with Dependabot directly.

This is possible through a workaround as Dependabot **can** update Dockerfiles.

To have Dependabot manage the images in a docker-compose configuration, simply:

1. Create a new subdirectory in your repository where you'd like to manage these images. I will use `docker/`
2. Create a Dockerfile for each of the services' images (use [`build`](https://docs.docker.com/compose/compose-file/compose-file-v3/#build) instead of [`image`](https://docs.docker.com/compose/compose-file/compose-file-v3/#image) in each service)
3. Configure your `.github/dependabot.yml` to check for `docker` updates

## Example

```yml
docker-compose.yml
---
version: '3'
services:
  pyvista:
    build:
      context: ./docker
      dockerfile: pyvista.Dockerfile
    ports:
      - 8000:8000
```

```Dockerfile
docker/pyvista.Dockerfile
---
FROM ghcr.io/pyvista/pyvista:v0.33.2
```

```yml
.github/dependabot.yml
---
version: 2
updates:
  - package-ecosystem: "docker"
    directory: "docker"
    schedule:
      interval: "daily"
    labels:
      - "dependencies"
```


Then, you will see Dependabot bump that image/service directly in the Dockerfile

![Screen Shot 2022-03-08 at 1 09 37 PM](https://user-images.githubusercontent.com/22067021/157316809-5032a2e6-0879-4ca7-858c-6d1bc323c67d.png)

![Screen Shot 2022-03-08 at 1 10 17 PM](https://user-images.githubusercontent.com/22067021/157316895-cd3ba0f3-9073-4b38-a579-c7a26358936b.png)



Demo in https://github.com/banesullivan/test-actions/pull/15 and example in https://github.com/banesullivan/test-actions/tree/1c7b0d3eab0e1a3b1da88153d0427a9b9442a726

**Comment by jokay** [comment]
> This is possible through a workaround as Dependabot **can** update Dockerfiles.

I think the goal should be to update a `docker-compose.yml` file directly without having to create (and maintain) additional Dockerfiles.

Even if Dependabot can update Dockerfiles it still cannot update `docker-compose.yml` files as of now.

**Comment by banesullivan** [comment]
> Even if Dependabot can update Dockerfiles it still cannot update docker-compose.yml files as of now (and this what this issue is about).

Exactly. I needed to have this working recently regardless of the state of this issue as it's clear in the thread above that this isn't changing anytime soon. Take my post above as merely a workaround with the current state of Dependabot for those who need something more immediately without switching to a different toolkit altogether. 

I look forward to this one day being addressed in full.

**Comment by kevinquillen** [comment]
> I think the goal should be to update a docker-compose.yml file directly without having to create (and maintain) additional Dockerfiles.
> 
> Even if Dependabot can update Dockerfiles it still cannot update docker-compose.yml files as of now (and this what this issue is about).

+1. I just landed here after getting this exact issue working in Renovate.

**Comment by AtzeDeVries** [comment]
For the work around also note you need to run a `docker compose build` before you actually on the newer image. You can run a `docker compose up --build` as a fallback. But we actually use a `docker compose run` which does not have the `build` flag.

**Comment by punkratz312** [comment]
would be awesome to have this feature

<img width="261" alt="image" src="https://user-images.githubusercontent.com/8830888/232319929-a6f0da00-0692-4b33-9267-04e7e4b01bb8.png">

i was wondering why its not working as it seems to be possible and its very common.

**Comment by jeffwidman** [comment]
To provide a little more clarity here, this _is_ something I'd personally like to see us eventually support. However, Docker as an ecosystem is complicated because there's a _lot_ of places that folks want us to bump docker images, see examples here:
* https://github.com/dependabot/dependabot-core/issues/7189

Although we shipped support for bumping docker image tags in k8s config files, I personally feel that was a bit of a mistake--I've already seen places in the code where we have to guess "is it a k8s file?" and we don't always get that right... so before we add support for bumping docker image tags in more places, I'd much rather see us cleanup how users can specify which type of language files they want us to bump docker images in.

So I think #7189 or a variant of it is what's needed to unblock this (as well as the other types of files users want to be able to bump docker image tags).

We know that the [current config file schema has some limitations](https://github.com/dependabot/dependabot-core/issues?q=is%3Aissue+is%3Aopen+label%3A%22F%3A+configuration-file%22), and while nothing is currently planned, it is a topic we'd like to re-examine in the not too distant future.

**Comment by webash** [comment]
but aren't Docker Compose files supposed to conform to `docker-compose(\..+)?\.ya?ml`? Even if not everyone is using them that way, if at least the base standard for Docker Compose files is implemented that's at least something? Then any future work to ensure file targeting is configurable can simply build on what's already there, rather than excluding Compose files indefinitely until the nirvana can be reached.

**Comment by slagiewka** [comment]
> but aren't Docker Compose files supposed to conform to `docker-compose(\..+)?\.ya?ml`?

Not exactly. [Since Compose v2](https://github.com/compose-spec/compose-spec/blob/853d1a9b2301c20b76b867a22b4ef0faef79362e/spec.md?plain=1#L163), `compose.yaml` is preferred. [Renovate uses](https://docs.renovatebot.com/modules/manager/docker-compose/#file-matching) `(^|/)(?:docker-)?compose[^/]*\.ya?ml$`.

**Comment by ab77** [comment]
Just buy Renovate already, integrate and call it done people 🤣

**Comment by sbe-arg** [comment]
I made this a few months ago as a workaround until is supported by dependabot. https://github.com/sbe-arg/simple-compose-service-updates in case anyone is interested

**Comment by powerman** [comment]
> I made this a few months ago as a workaround until is supported by dependabot. https://github.com/sbe-arg/simple-compose-service-updates in case anyone is interested

Looks interesting, but these two points sounds too limiting: "compose files must be on your repo root" (complex projects often have several compose files in different dirs) and "requires full registry including default docker.io/…" (no one name images from default registry this way in compose files and requiring this won't work because people will forget to do this because compose itself will work without this and only updates will be broken which is much harder to notice).

**Comment by sbe-arg** [comment]
The compose file locations is a very easy fix, noone has requested a path var or a full scan of yaml files in all subdirectories.

The registry names is harder although for docker hub it can be mocked, other registries always have to be specified.

Open issues and ill try to address them. Thats the normal process for open sourcing.

**Comment by njfamirm** [comment]
any update?

> 👍 for this, and should be relatively straightforward. A couple of things I want to get to first, but I'm definitely game for adding this to Dependabot at some point!

**Comment by AliMD** [comment]
whats up?
or any temporary solution after 5 years!

**Comment by mountainash** [comment]
[Here's a pointer](https://github.com/renovatebot/renovate/tree/main/lib/modules/manager/docker-compose) to the Renovate code that could jump-start a Developer in to making this work on Dependabot.

**Comment by njfamirm** [comment]
> > Any temporary solution
> 
> @AliMD, I recommend reading through all of the above comments for a few different workarounds. My comment above is, in my opinion, a fairly robust workaround: [#390 (comment)](https://github.com/dependabot/dependabot-core/issues/390#issuecomment-1062170379)

thank's
this can solve as a tmp solution

**Comment by njfamirm** [comment]
> [Here's a pointer](https://github.com/renovatebot/renovate/tree/main/lib/modules/manager/docker-compose) to the Renovate code that could jump-start a Developer in to making this work on Dependabot.

@mountainash Thank you, but we currently use Dependabot. Switching to another tool would involve a time investment.

**Comment by njfamirm** [comment]
> 👍 for this, and should be relatively straightforward. A couple of things I want to get to first, but I'm definitely game for adding this to Dependabot at some point!

@greysteil is any plan for support this? or use another tool?

**Comment by greysteil** [comment]
Not sure - I haven't worked on Dependabot for a few years now. @jeffwidman might know?

**Comment by jeffwidman** [comment]
Unfortunately I'm also no longer on the Dependabot maintainer team so have no additional insight into this.

**Comment by kevinquillen** [comment]
The only solution for me was just using Renovate for everything, either self hosted or as the free GitHub App.

**Comment by marchershey** [comment]
So I'm guessing we're all jumping ship and boarding Renovate? :P

**Comment by PythonCoderAS** [comment]
Is this still planned?

**Comment by aaiezza** [comment]
Would definitely love this!
I have the PR ready to go 🙂

```docker-compose
version: 2
updates:
  - package-ecosystem: "maven"
    directory: "/"
    schedule:
      interval: "daily"

  - package-ecosystem: "docker"
    directory: "/"
    schedule:
      interval: "weekly"

  - package-ecosystem: "docker-compose"
    directory: "/"
    schedule:
      interval: "weekly"

  - package-ecosystem: "docker"
    directory: "/kubernetes/"
    schedule:
      interval: "weekly"

  - package-ecosystem: "docker"
    directory: "/postman/"
    schedule:
      interval: "weekly"
```

**Comment by abdulapopoola** [comment]
Hello everyone,

I apologize for the lack of engagement and have some good news to share. Working on `docker-compose` is on our roadmap for this quarter (Jan - Mar, 2025). We're still finetuning our plans and hope to eventually publish a public roadmap to make this easier.

Thanks for all the help and support!

**Comment by abdulapopoola** [comment]
Another update, we expect work to start on this in February. @robaiken will be the developer driving this 😄

**Comment by robaiken** [comment]
I have started working on adding Docker Compose support to Dependabot Core! You can follow the development progress here: https://github.com/dependabot/dependabot-core/pull/11492

**Comment by robaiken** [comment]
The Docker Compose updater is now available in beta! To enable it, add the following to your `dependabot.yml` configuration:

```

enable-beta-ecosystems: true
updates:
  - package-ecosystem: "docker-compose" 
    directory: "/" # Location of package manifests
```

**Important notes:**

* Make sure to include enable-beta-ecosystems: true in your config, otherwise the Docker Compose updater won't be activated
* Please reply to this comment if you encounter any errors or issues during the beta period - your feedback helps us improve!

**Comment by ulgens** [comment]
Tested on https://github.com/ulgens/django-blasphemy and it seems to be working fine: https://github.com/ulgens/django-blasphemy/actions/runs/13437473835

**Comment by swissbuechi** [comment]
@robaiken 
Awesome, thanks!

Worked for me: https://github.com/swissbuechi/one-time-secret/pull/84/files

**Comment by Bert-R** [comment]
@robaiken Thanks for fixing this! It works for me on a whole set of repositories but it fails for Elasticsearch, Kibana and Filebeat and also for NGINX. Here is the log for Elasticsearch:

```
Run github/dependabot-action@main
🤖 ~ starting update ~
Fetching job details
Pulling updater images
Starting update process
Created proxy container: f9[8](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:9)28a79e460d0d9e759296408e5763cca35c1dabe3c81c8b87b1a5870d7a4[9](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:10)8
Created container: 0aa7dd1e08dcf50ed8a882919db8af02409244b85a64712ef94d7ae8ca430728
  proxy | 2025/02/20 14:35:58 proxy starting, commit: 50bb55ecec5571c86c9d56b665965ae41b6382a7
  proxy | 2025/02/20 14:35:58 Listening (:[10](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:11)80)
updater | Updating certificates in /etc/ssl/certs...
updater | rehash: warning: skipping ca-certificates.crt,it does not contain exactly one certificate or CRL
updater | 1 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
updater | done.
updater | 2025/02/20 14:36:01 INFO <job_968051222> Starting job processing
2025/02/20 14:36:01 INFO <job_968051222> Job definition: {"job":{"allowed-updates":[{"dependency-type":"direct","update-type":"all"}],"commit-message-options":{"prefix":null,"prefix-development":null,"include-scope":null},"credentials-metadata":[{"type":"git_source","host":"github.com"}],"debug":null,"dependencies":null,"dependency-groups":[],"dependency-group-to-refresh":null,"existing-pull-requests":[],"existing-group-pull-requests":[],"experiments":{"record-ecosystem-versions":true,"record-update-job-unknown-error":true,"proxy-cached":true,"move-job-token":true,"dependency-change-validation":true,"nuget-install-dotnet-sdks":true,"nuget-native-analysis":true,"nuget-use-direct-discovery":true,"enable-file-parser-python-local":true,"npm-fallback-version-above-v6":true,"npm-v6-deprecation-warning":true,"npm-v6-unsupported-error":true,"lead-security-dependency":true,"enable-record-ecosystem-meta":true,"enable-shared-helpers-command-timeout":true,"enable-beta-ecosystems":true,"enable-engine-version-detection":true,"avoid-duplicate-updates-package-json":true,"allow-refresh-for-existing-pr-dependencies":true,"enable-bun-ecosystem":true,"exclude-local-composer-packages":true},"ignore-conditions":[],"lockfile-only":false,"max-updater-run-time":2700,"package-manager":"docker_compose","proxy-log-response-body-on-auth-failure":true,"requirements-update-strategy":null,"reject-external-code":false,"security-advisories":[],"security-updates-only":false,"source":{"provider":"github","repo":"Bert-R/container-setup","branch":null,"directory":"/elasticsearch","api-endpoint":"https://api.github.com/","hostname":"github.com"},"updating-a-pull-request":false,"update-subdependencies":false,"vendor-dependencies":false,"enable-beta-ecosystems":true,"repo-private":true}}
updater | 2025/02/20 14:36:01 INFO <job_968051222> Started process PID: 1070 with command: {} git config --global credential.helper '!/home/dependabot/common/lib/dependabot/../../bin/git-credential-store-immutable --file /home/dependabot/dependabot-updater/git.store' {}
updater | 2025/02/20 14:36:01 INFO <job_968051222> Process PID: 1070 completed with status: pid 1070 exit 0
updater | 2025/02/20 14:36:01 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:01 INFO <job_968051222> Started process PID: 1077 with command: {} git config --global --replace-all url.https://github.com/.insteadOf ssh://git@github.com/ {}
updater | 2025/02/20 14:36:01 INFO <job_968051222> Process PID: 1077 completed with status: pid 1077 exit 0
updater | 2025/02/20 14:36:01 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:01 INFO <job_968051222> Started process PID: 1083 with command: {} git config --global --add url.https://github.com/.insteadOf ssh://git@github.com: {}
updater | 2025/02/20 14:36:01 INFO <job_968051222> Process PID: 1083 completed with status: pid 1083 exit 0
updater | 2025/02/20 14:36:01 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:01 INFO <job_968051222> Started process PID: 1089 with command: {} git config --global --add url.https://github.com/.insteadOf git@github.com: {}
updater | 2025/02/20 14:36:01 INFO <job_968051222> Process PID: 1089 completed with status: pid 1089 exit 0
updater | 2025/02/20 14:36:01 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:01 INFO <job_968051222> Started process PID: 1095 with command: {} git config --global --add url.https://github.com/.insteadOf git@github.com/ {}
updater | 2025/02/20 14:36:01 INFO <job_968051222> Process PID: 1095 completed with status: pid 1095 exit 0
updater | 2025/02/20 14:36:01 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:01 INFO <job_968051222> Started process PID: 1101 with command: {} git config --global --add url.https://github.com/.insteadOf git://github.com/ {}
updater | 2025/02/20 14:36:01 INFO <job_968051222> Process PID: 1101 completed with status: pid 1101 exit 0
updater | 2025/02/20 14:36:01 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:01 INFO <job_968051222> Started process PID: 1107 with command: {} git clone --no-tags --depth 1 --recurse-submodules --shallow-submodules https://github.com/Bert-R/container-setup /home/dependabot/dependabot-updater/repo {}
  proxy | 2025/02/20 14:36:01 [002] GET https://github.com:443/Bert-R/container-setup/info/refs?service=git-upload-pack
  proxy | 2025/02/20 14:36:01 [002] * authenticating git server request (host: github.com)
  proxy | 2025/02/20 14:36:01 [002] 200 https://github.com:443/Bert-R/container-setup/info/refs?service=git-upload-pack
  proxy | 2025/02/20 14:36:01 [004] POST https://github.com:443/Bert-R/container-setup/git-upload-pack
2025/02/20 14:36:01 [004] * authenticating git server request (host: github.com)
  proxy | 2025/02/20 14:36:01 [004] 200 https://github.com:443/Bert-R/container-setup/git-upload-pack
  proxy | 2025/02/20 14:36:02 [006] POST https://github.com:443/Bert-R/container-setup/git-upload-pack
2025/02/20 14:36:02 [006] * authenticating git server request (host: github.com)
  proxy | 2025/02/20 14:36:02 [006] 200 https://github.com:443/Bert-R/container-setup/git-upload-pack
updater | 2025/02/20 14:36:02 INFO <job_968051222> Process PID: 1107 completed with status: pid 1107 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.7 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1[15](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:17)0 with command: {} git -C /home/dependabot/dependabot-updater/repo ls-files --stage {}
updater | 2025/02/20 14:36:02 INFO <job_968051222> Process PID: 1150 completed with status: pid 1150 exit 0
2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1[16](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:18)3 with command: {} git config --global credential.helper '!/home/dependabot/common/lib/dependabot/../../bin/git-credential-store-immutable --file /home/dependabot/dependabot-updater/git.store' {}
updater | 2025/02/20 14:36:02 INFO <job_968051222> Process PID: 1163 completed with status: pid 1163 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1[17](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:19)0 with command: {} git config --global --replace-all url.https://github.com/.insteadOf ssh://git@github.com/ {}
updater | 2025/02/20 14:36:02 INFO <job_968051222> Process PID: 1170 completed with status: pid 1170 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1176 with command: {} git config --global --add url.https://github.com/.insteadOf ssh://git@github.com: {}
updater | 2025/02/20 14:36:02 INFO <job_968051222> Process PID: 1176 completed with status: pid 1176 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1[18](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:20)2 with command: {} git config --global --add url.https://github.com/.insteadOf git@github.com: {}
updater | 2025/02/20 14:36:02 INFO <job_968051222> Process PID: 1182 completed with status: pid 1182 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1188 with command: {} git config --global --add url.https://github.com/.insteadOf git@github.com/ {}
updater | 2025/02/20 14:36:02 INFO <job_968051222> Process PID: 1188 completed with status: pid 1188 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1[19](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:21)4 with command: {} git config --global --add url.https://github.com/.insteadOf git://github.com/ {}
updater | [20](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:22)25/02/20 14:36:02 INFO <job_968051222> Process PID: 1194 completed with status: pid 1194 exit 0
2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1200 with command: {} git rev-parse HEAD {}
updater | 2025/02/20 14:36:02 INFO <job_968051222> Process PID: 1200 completed with status: pid 1200 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1[21](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:23)3 with command: {} git config --global credential.helper '!/home/dependabot/common/lib/dependabot/../../bin/git-credential-store-immutable --file /home/dependabot/dependabot-updater/git.store' {}
updater | 2025/02/20 14:36:02 INFO <job_968051[22](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:24)2> Process PID: 1213 completed with status: pid 1213 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1220 with command: {} git config --global --replace-all url.https://github.com/.insteadOf ssh://git@github.com/ {}
updater | 2025/02/20 14:36:02 INFO <job_968051222> Process PID: 1220 completed with status: pid 1220 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1226 with command: {} git config --global --add url.https://github.com/.insteadOf ssh://git@github.com: {}
updater | 2025/02/20 14:36:02 INFO <job_968051222> Process PID: 1226 completed with status: pid 1226 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1[23](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:25)2 with command: {} git config --global --add url.https://github.com/.insteadOf git@github.com: {}
updater | 2025/02/20 14:36:02 INFO <job_968051222> Process PID: 1232 completed with status: pid 1232 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1238 with command: {} git config --global --add url.https://github.com/.insteadOf git@github.com/ {}
updater | 2025/02/20 14:36:02 INFO <job_968051222> Process PID: 1238 completed with status: pid 1238 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1[24](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:26)4 with command: {} git config --global --add url.https://github.com/.insteadOf git://github.com/ {}
updater | 20[25](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:27)/02/20 14:36:02 INFO <job_968051222> Process PID: 1244 completed with status: pid 1244 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Started process PID: 1250 with command: {} git rev-parse HEAD {}
updater | 2025/02/20 14:36:02 INFO <job_968051222> Process PID: 1250 completed with status: pid 1250 exit 0
updater | 2025/02/20 14:36:02 INFO <job_968051222> Total execution time: 0.01 seconds
updater | 2025/02/20 14:36:02 INFO <job_968051222> Base commit SHA: 847f373a4e4a09baa0ace459cb27d59427708e6f
updater | 2025/02/20 14:36:02 INFO <job_968051222> Finished job processing
updater | 2025/02/20 14:36:04 INFO <job_968051222> Starting job processing
  proxy | 2025/02/20 14:36:04 [008] POST /update_jobs/968051222/update_dependency_list
  proxy | 2025/02/20 14:36:04 [008] 204 /update_jobs/968051222/update_dependency_list
  proxy | 2025/02/20 14:36:04 [010] POST /update_jobs/968051222/increment_metric
  proxy | 2025/02/20 14:36:04 [010] 204 /update_jobs/968051222/increment_metric
updater | 2025/02/20 14:36:04 INFO <job_968051222> Starting update job for Bert-R/container-setup
2025/02/20 14:36:04 INFO <job_968051222> Checking all dependencies for version updates...
updater | 2025/02/20 14:36:04 INFO <job_968051222> Checking if elastic/elasticsearch 8.16.4 needs updating
  proxy | 2025/02/20 14:36:04 [012] GET https://registry.hub.docker.com:443/v2/elastic/elasticsearch/tags/list
  proxy | 2025/02/20 14:36:04 [012] 401 https://registry.hub.docker.com:443/v2/elastic/elasticsearch/tags/list
  proxy | 2025/02/20 14:36:05 [014] GET https://auth.docker.io:443/token?service=registry.docker.io&scope=repository%3Aelastic%2Felasticsearch%3Apull&account
  proxy | 2025/02/20 14:36:05 [014] 200 https://auth.docker.io:443/token?service=registry.docker.io&scope=repository%3Aelastic%2Felasticsearch%3Apull&account
  proxy | 2025/02/20 14:36:05 [016] GET https://registry.hub.docker.com:443/v2/elastic/elasticsearch/tags/list
  proxy | 2025/02/20 14:36:05 [016] 200 https://registry.hub.docker.com:443/v2/elastic/elasticsearch/tags/list
updater | 2025/02/20 14:36:05 INFO <job_968051222> Latest version is 8.17.2
updater | 2025/02/20 14:36:05 INFO <job_968051222> Requirements to unlock own
2025/02/20 14:36:05 INFO <job_968051222> Requirements update strategy 
updater | 2025/02/20 14:36:05 INFO <job_968051222> Updating elastic/elasticsearch from 8.16.4 to 8.17.2
  proxy | 2025/02/20 14:36:05 [018] POST /update_jobs/968051222/record_update_job_unknown_error
  proxy | 2025/02/20 14:36:05 [018] 204 /update_jobs/968051222/record_update_job_unknown_error
  proxy | 2025/02/20 14:36:05 [020] POST /update_jobs/968051222/record_update_job_error
  proxy | 2025/02/20 14:36:05 [020] 204 /update_jobs/968051222/record_update_job_error
  proxy | 2025/02/20 14:36:05 [022] POST /update_jobs/968051222/increment_metric
  proxy | 2025/02/20 14:36:05 [022] 204 /update_jobs/968051222/increment_metric
  proxy | 2025/02/20 14:36:05 [024] POST /update_jobs/968051222/record_update_job_unknown_error
  proxy | 2025/02/20 14:36:05 [024] 204 /update_jobs/968051222/record_update_job_unknown_error
updater | 2025/02/20 14:36:05 ERROR <job_968051222> Error processing elastic/elasticsearch (RuntimeError)
2025/02/20 14:36:05 ERROR <job_968051222> Expected content to change!
updater | 2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/docker_compose/lib/dependabot/shared/shared_file_updater.rb:64:in `updated_dockerfile_content'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `bind_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/docker_compose/lib/dependabot/docker_compose/file_updater.rb:48:in `block in updated_dependency_files'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/docker_compose/lib/dependabot/docker_compose/file_updater.rb:43:in `each'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/docker_compose/lib/dependabot/docker_compose/file_updater.rb:43:in `updated_dependency_files'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `bind_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/dependency_change_builder.rb:143:in `generate_dependency_files'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `bind_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/dependency_change_builder.rb:72:in `run'
updater | 2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `bind_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
updater | 2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
updater | 2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/dependency_change_builder.rb:44:in `create_from'
updater | 2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `bind_call'
updater | 2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/update_all_versions.rb:170:in `check_and_create_pull_request'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `bind_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/update_all_versions.rb:94:in `check_and_create_pr_with_error_handling'
updater | 2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `bind_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
updater | 2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/update_all_versions.rb:59:in `block in perform'
updater | 2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/update_all_versions.rb:59:in `each'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/update_all_versions.rb:59:in `perform'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `bind_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/group_update_all_versions.rb:137:in `block in run_ungrouped_dependency_updates'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/group_update_all_versions.rb:122:in `each'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/group_update_all_versions.rb:122:in `run_ungrouped_dependency_updates'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `bind_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/updater/operations/group_update_all_versions.rb:65:in `perform'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `bind_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/updater.rb:45:in `run'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/update_files_command.rb:45:in `block in perform_job'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/opentelemetry-api-1.4.0/lib/opentelemetry/trace/tracer.rb:37:in `block in in_span'
updater | 2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/opentelemetry-api-1.4.0/lib/opentelemetry/trace.rb:70:in `block in with_span'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/opentelemetry-api-1.4.0/lib/opentelemetry/context.rb:87:in `with_value'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/opentelemetry-api-1.4.0/lib/opentelemetry/trace.rb:70:in `with_span'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/opentelemetry-api-1.4.0/lib/opentelemetry/trace/tracer.rb:37:in `in_span'
updater | 2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/update_files_command.rb:18:in `perform_job'
2025/02/20 14:36:05 ERROR <job_968051222> /home/dependabot/dependabot-updater/lib/dependabot/base_command.rb:37:in `run'
2025/02/20 14:36:05 ERROR <job_968051222> bin/update_files.rb:46:in `<main>'
  proxy | 2025/02/20 14:36:05 [0[26](https://github.com/Bert-R/container-setup/runs/37543951143?check_suite_focus=true#step:3:28)] POST /update_jobs/968051222/record_ecosystem_meta
  proxy | 2025/02/20 14:36:05 [026] 204 /update_jobs/968051222/record_ecosystem_meta
  proxy | 2025/02/20 14:36:05 [028] PATCH /update_jobs/968051222/mark_as_processed
  proxy | 2025/02/20 14:36:05 [028] 204 /update_jobs/968051222/mark_as_processed
updater | 2025/02/20 14:36:05 INFO <job_968051222> Finished job processing
updater | 2025/02/20 14:36:05 INFO Results:
Dependabot encountered '1' error(s) during execution, please check the logs for more details.
+---------------------------------------+
|     Dependencies failed to update     |
+-----------------------+---------------+
| elastic/elasticsearch | unknown_error |
+-----------------------+---------------+
Failure running container 0aa7dd1e08dcf50ed8a882919db8af02409244b85a64712ef94d7ae8ca430728
Cleaned up container 0aa7dd1e08dcf50ed8a882919db8af02409244b85a64712ef94d7ae8ca430728
  proxy | 2025/02/20 14:36:05 Posting metrics to remote API endpoint
2025/02/20 14:36:05 0/14 calls cached (0%)
Error: Dependabot encountered an error performing the update

Error: The updater encountered one or more errors.

For more information see: https://github.com/Bert-R/container-setup/network/updates/968051222 (write access to the repository is required to view the log)
🤖 ~ finished: error reported to Dependabot ~
```

**Comment by krokyze** [comment]
@robaiken Amazing work! Just tested it, and it works flawlessly. Do you think Dependabot could also support bumping versions defined this way?

```
roundcubemail:
  build:
    dockerfile_inline: |
      FROM roundcube/roundcubemail:1.6.8-apache
      RUN ln -sf /dev/null /var/log/apache2/access.log && \
          ln -sf /dev/null /var/log/apache2/other_vhosts_access.log
```

**Comment by Aaron-Ritter** [comment]
thanks a lot @robaiken for taking this on!

I was able to successfully tests 10 different compose files so far, with the exception of this from the novu project which is failing:

**My primary issue** is this example (In this case a fork of their example, but we have the same error in other internal repos):
https://github.com/sonderformat-llc/novu/actions/runs/13439797695/job/37551070482
- Trying to update https://github.com/sonderformat-llc/novu/tree/next/docker/community

**Secondary** is just something i discovered out of curiosity testing some other docker-compose configuration in their repo:
https://github.com/sonderformat-llc/novu/actions/runs/13439797734/job/37551070374

**Comment by robaiken** [comment]
@gegoune Thanks for reporting this issue. I've added support for YAML with aliases.

@krokyze Thanks for asking about this feature. This was an oversight on my part - I'll add support for `dockerfile_inline` image versions. Please let me know how it works for you.

**Comment by krokyze** [comment]
@robaiken wow that was quick! Just gave it a try and `dockerfile_inline` bumps as expected! Thanks!

**Comment by alimd** [comment]
> The Docker Compose updater is now available in beta! To enable it, add the following to your `dependabot.yml` configuration:
> 
> ```
> 
> enable-beta-ecosystems: true
> updates:
>   - package-ecosystem: "docker-compose" 
>     directory: "/" # Location of package manifests
> ```
> 
> **Important notes:**
> 
> * Make sure to include enable-beta-ecosystems: true in your config, otherwise the Docker Compose updater won't be activated
> * Please reply to this comment if you encounter any errors or issues during the beta period - your feedback helps us improve!

It’s not shown under “Dependency graph,” and I can’t manually test it.

**Comment by DigitallyRefined** [comment]
@robaiken thanks for adding `docker_compose` support! 

>  *  `sha256:` string was removed in updated image version

I think this looks like a bug to me, as I sometimes have to use the `sha256` to make sure I get the right version and some container images only have a `latest` tag (like [alpinelinux/unbound](https://hub.docker.com/r/alpinelinux/unbound/tags)), which gave me this Dependabot PR:

```patch
-    image: alpinelinux/unbound@sha256:c1254bc07c8b4ad59a526f156b25cf41c71e5c143170dda325408aaa69a6da61
+    image: alpinelinux/unbound@5d6daa0b4d4d28408dbb584c6fef8a21d2cc20b86838a7fb3c1ff33e4a9932df
```

Removing the `sha256:` results in a `invalid reference format` error when I try and pull it:

```bash
docker compose pull
unable to get image 'alpinelinux/unbound@5d6daa0b4d4d28408dbb584c6fef8a21d2cc20b86838a7fb3c1ff33e4a9932df': Error response from daemon: invalid reference format
exit status 1
```

**Comment by gmile** [comment]
Appears to be failing for me 🤔 Attaching a snippet with a stacktrace I just saw in GH Actions:

```
Dependabot encountered '1' error(s) during execution, please check the logs for more details.
```

<details>

```
updater | 2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/docker_compose/lib/dependabot/docker_compose/file_parser.rb:41:in `block (2 levels) in parse'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/docker_compose/lib/dependabot/docker_compose/file_parser.rb:38:in `each'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/docker_compose/lib/dependabot/docker_compose/file_parser.rb:38:in `block in parse'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/docker_compose/lib/dependabot/docker_compose/file_parser.rb:36:in `each'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/docker_compose/lib/dependabot/docker_compose/file_parser.rb:36:in `parse'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `bind_call'
updater | 2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/lib/dependabot/dependency_snapshot.rb:270:in `parse_files!'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.[115](https://github.com/Contractbook/danlon-e2e/actions/runs/13454996557/job/37596964201#step:3:117)77/lib/types/private/methods/call_validation.rb:270:in `bind_call'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/lib/dependabot/dependency_snapshot.rb:230:in `block in initialize'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/lib/dependabot/dependency_snapshot.rb:228:in `each'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/lib/dependabot/dependency_snapshot.rb:228:in `initialize'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `bind_call'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/lib/dependabot/dependency_snapshot.rb:38:in `new'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/lib/dependabot/dependency_snapshot.rb:38:in `create_from_job_definition'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `bind_call'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/call_validation.rb:270:in `validate_call'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/sorbet-runtime-0.5.11577/lib/types/private/methods/_methods.rb:277:in `block in _on_method_added'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/lib/dependabot/update_files_command.rb:22:in `block in perform_job'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/opentelemetry-api-1.4.0/lib/opentelemetry/trace/tracer.rb:37:in `block in in_span'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/opentelemetry-api-1.4.0/lib/opentelemetry/trace.rb:70:in `block in with_span'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/opentelemetry-api-1.4.0/lib/opentelemetry/context.rb:87:in `with_value'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/opentelemetry-api-1.4.0/lib/opentelemetry/trace.rb:70:in `with_span'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/vendor/ruby/3.3.0/gems/opentelemetry-api-1.4.0/lib/opentelemetry/trace/tracer.rb:37:in `in_span'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/lib/dependabot/update_files_command.rb:18:in `perform_job'
2025/02/21 10:34:52 ERROR <job_968517285> /home/dependabot/dependabot-updater/lib/dependabot/base_command.rb:37:in `run'
2025/02/21 10:34:52 ERROR <job_968517285> bin/update_files.rb:46:in `<main>'
  proxy | 2025/02/21 10:34:52 [008] POST /update_jobs/968517285/record_update_job_unknown_error
  proxy | 2025/02/21 10:34:52 [008] 204 /update_jobs/968517285/record_update_job_unknown_error
  proxy | 2025/02/21 10:34:52 [010] POST /update_jobs/968517285/record_update_job_error
  proxy | 2025/02/21 10:34:53 [010] 204 /update_jobs/968517285/record_update_job_error
  proxy | 2025/02/21 10:34:53 [012] POST /update_jobs/968517285/record_update_job_unknown_error
  proxy | 2025/02/21 10:34:53 [012] 204 /update_jobs/968517285/record_update_job_unknown_error
  proxy | 2025/02/21 10:34:53 [014] PATCH /update_jobs/968517285/mark_as_processed
  proxy | 2025/02/21 10:34:53 [014] 204 /update_jobs/968517285/mark_as_processed
updater | 2025/02/21 10:34:53 INFO <job_968517285> Finished job processing
updater | 2025/02/21 10:34:53 INFO Results:
Dependabot encountered '1' error(s) during execution, please check the logs for more details.
```
</details>

**Comment by robaiken** [comment]
@DigitallyRefined, @gegoune Thank you both for reporting these issues. I've fixed the digest handling in the Docker Compose updater. @gegoune - the PR title will now look something like this:
```
Bump ubuntu from `1830542` to `7229784` in /
```
Please let me know if you notice any other issues!

**Comment by gegoune** [comment]
@robaiken It works better now, thanks. Unfortunately it is still impossible to tell what's going on.

I don't think registries allow searching by digest (at least Docker Hub doesn't seem to), meaning that it's impossible or at least very difficult to tell which version PR updates to with just:
```
build(deps): bump postgres from `a35ec42` to `0321e22`
```

Can human readable versions be used instead?

Also, to bring my other point from previous comment. It's common practice to keep human-readable version string in comment after digest. I would like to see that string updated as well, otherwise we run into issue as described in first paragraph again.

So,

```
image: postgres@sha256:a35ec42526e3c522eb13b4d82eddaee875d0ac6ca9eb5cc5607e412854478c71 # 16.7
```
to be updated to
```
image: postgres@sha256:{new hash} # 16.8
```
(theoretical version)

**Comment by robaiken** [comment]
@gegoune it's definitely a limitation with how things work right now. The version readability stuff is a bit outside what we can tackle in this implementation.

Mind opening up a feature request for this? Would be great to track things like showing readable versions and keeping those comments in sync. It'll help us plan this out properly for the future.

**Comment by Aaron-Ritter** [comment]
> Appears to be failing for me 🤔 Attaching a snippet with a stacktrace I just saw in GH Actions:
> 
> ```
> Dependabot encountered '1' error(s) during execution, please check the logs for more details.
> ```
> 
> For what it's worth, some of the images in my `docker-compose.yml` are "parametrized", e.g. look like this:
> 
> services:
>   api:
>     image: ${E2E_API_IMAGE_TAG}
>     pull_policy: missing
>     # ...

@robaiken FYI in case you are planning to address env param in the future. I had the same issue with a test, which had mostly parameterized image entries with the exception of one "hard-coded". In my example it wouldn't have been able to lookup the ENV as it is generated ad-hoc with test scripts, but my plan was to still use dependabot to update the "hard-coded" ones:
https://github.com/sonderformat-llc/Mailu/actions/runs/13440943004/job/37554951541

Good news, the new version with [#11656](https://github.com/dependabot/dependabot-core/pull/11656), works for the "hard-coded" image entry and skips the rest:
https://github.com/sonderformat-llc/Mailu/actions/runs/13462353541/job/37620450101

**Comment by Bert-R** [comment]
@robaiken Could it be that you only read the first part of https://github.com/dependabot/dependabot-core/issues/390#issuecomment-2671717979 ("Thanks for fixing this! It works for me on a whole set of repositories")? I see a :tada: reaction, but I also reported an issue  ("but it fails for Elasticsearch, Kibana and Filebeat and also for NGINX."). The log is included in that comment. I just retested and it still fails.

The compose file is trivially simple:
```
services:
  elasticsearch:
    image: "elastic/elasticsearch:8.16.4"
    container_name: "elasticsearch"
    volumes:
      - "/.../data:/usr/share/elasticsearch/data"
      - "/.../config/elasticsearch.yml:/usr/share/elasticsearch/config/elasticsearch.yml"
    ports:
      - "9200:9200"
      - "9300:9300"
    network_mode: "bridge"
    environment:
      - "ES_JAVA_OPTS=-Xms4g -Xmx4g"
    deploy:
      resources:
        limits:
          memory: 4GB
    restart: "always"
```

The compose files for Filebeat and Kibana have the same structure and reference the same version number.

For NGINX, this is the compose file:
```
services:
    nginx:
        image: "nginx:1.27.2"
        container_name: nginx
        volumes:
            - "/.../config/nginx.conf:/etc/nginx/nginx.conf"
            - "/.../config/conf.d:/etc/nginx/conf.d"
            - "/.../config/extra:/etc/nginx/extra"
            - "/.../certs:/etc/ssl/private"
        ports:
            - "80:80"
            - "443:443"
        network_mode: "bridge"
        extra_hosts:
            - "host.docker.internal:host-gateway"
        restart: always
```

**Comment by gmile** [comment]
Looks like dependabot will refuse to upgrade an image with the following tag:

```yaml
service:
  serviceA:
    image: hexpm/elixir:1.18.2-erlang-27.2.2-alpine-3.21.2
```

However, today the following images are already available:

```
hexpm/elixir:1.18.2-erlang-27.2.3-alpine-3.21.3
hexpm/elixir:1.18.2-erlang-27.2.4-alpine-3.21.3
```

For reference, these are official images built & published by Elixir / Hexpm team, see https://github.com/hexpm/bob?tab=readme-ov-file#docker-images

**Comment by Aaron-Ritter** [comment]
@gmile this could be related to this https://github.com/dependabot/dependabot-core/issues/10678. 

And i had previously similar issues with updates in Dockerfile and k8s files, as soon i used a definition of `repository_name:semver-alpine-semver`, or for example with mongo-express where it suggested to update 1.0.2-18-alpine3.18 to 1-18-alpine3.18. Plus dependabot does as well some evaluation on where the latest tag is.

**Comment by robaiken** [comment]
@Aaron-Ritter Thanks for reporting this! The current implementation focuses only on hardcoded image tags, and I'm glad it's working for those in your setup.

For images using environment variables (like `${E2E_API_IMAGE_TAG}`), would you mind opening a feature request? This would be a separate feature requiring its own implementation.

**Comment by Aaron-Ritter** [comment]
@robaiken sure, will create a issue accordingly once i've updated everything with the official release.

With regards to the earlier reported issue https://github.com/dependabot/dependabot-core/issues/390#issuecomment-2672020542 on images not updating and trowing an error. 

This seems to be related to the quotation of the images, i updated my test with single and double quotes as well as without quotes, the one without is now updating propperly:
https://github.com/sonderformat-llc/novu/actions/runs/13496505546/job/37704897458

So the issue reported from @Bert-R seems to be as well related to the quotation as double quotes are used in this example:
https://github.com/dependabot/dependabot-core/issues/390#issuecomment-2676048313

**Comment by robaiken** [comment]
Thanks for the detailed investigation! Yes, you're correct - I've submitted PR https://github.com/dependabot/dependabot-core/pull/11669 to add support for images with quotes. This should resolve the issues you and @Bert-R encountered with quoted image tags.

**Comment by robaiken** [comment]
@gmile Could you share the job logs from one of your failed runs? You can find them following [GitHub's guide on viewing Dependabot logs](https://docs.github.com/en/code-security/dependabot/troubleshooting-dependabot/viewing-dependabot-job-logs).

**Comment by Aaron-Ritter** [comment]
@robaiken thanks a lot! I can confirm this now works for all my tests.

https://github.com/sonderformat-llc/novu/actions/runs/13498872364/job/37712208984
https://github.com/sonderformat-llc/novu/actions/runs/13498870760/job/37712205124

**Comment by gmile** [comment]
@robaiken sure. I get this at the end of logs:

```
2025-02-24T11:44:24.5799611Z updater | 2025/02/24 11:44:24 INFO <job_969523985> Finished job processing
2025-02-24T11:44:24.5810598Z updater | 2025/02/24 11:44:24 INFO Results:
2025-02-24T11:44:24.5813338Z Dependabot encountered '1' error(s) during execution, please check the logs for more details.
2025-02-24T11:44:24.5815455Z +-----------------------------------------------+
2025-02-24T11:44:24.5815963Z |         Dependencies failed to update         |
2025-02-24T11:44:24.5816455Z +--------------+--------------------------------+
2025-02-24T11:44:24.5816966Z | hexpm/elixir | dependency_file_not_resolvable |
2025-02-24T11:44:24.5817478Z +--------------+--------------------------------+
```

Can't share the entire log since the repo name is sensitive to share be shared publicly, but here are Elixir-related lines from it - this is what was logged before dependabot moved on checking another image. To sum up, it seems like its attempts to access registry.hub.docker.com are failing:

<details>
```
2025-02-24T11:43:51.7383374Z updater | 2025/02/24 11:43:51 INFO <job_969523985> Checking if hexpm/elixir 1.18.2-erlang-27.2.2-alpine-3.21.2 needs updating
2025-02-24T11:43:51.7704386Z   proxy | 2025/02/24 11:43:51 [236] GET https://registry.hub.docker.com:443/v2/hexpm/elixir/tags/list
2025-02-24T11:43:51.8076709Z   proxy | 2025/02/24 11:43:51 [236] 401 https://registry.hub.docker.com:443/v2/hexpm/elixir/tags/list
2025-02-24T11:43:51.8389915Z   proxy | 2025/02/24 11:43:51 [238] GET https://auth.docker.io:443/token?service=registry.docker.io&scope=repository%3Ahexpm%2Felixir%3Apull&account
2025-02-24T11:43:51.9669235Z   proxy | 2025/02/24 11:43:51 [238] 200 https://auth.docker.io:443/token?service=registry.docker.io&scope=repository%3Ahexpm%2Felixir%3Apull&account
2025-02-24T11:43:51.9979545Z   proxy | 2025/02/24 11:43:51 [240] GET https://registry.hub.docker.com:443/v2/hexpm/elixir/tags/list
2025-02-24T11:43:57.0077144Z   proxy | 2025/02/24 11:43:57 [242] GET https://registry.hub.docker.com:443/v2/hexpm/elixir/tags/list
2025-02-24T11:43:59.5757363Z   proxy | 2025/02/24 11:43:59 [240] 200 https://registry.hub.docker.com:443/v2/hexpm/elixir/tags/list
2025-02-24T11:43:59.5759220Z   proxy | 2025/02/24 11:43:59 [240] WARN: Cannot write TLS response header from mitm'd client: write tcp 172.19.0.2:1080->172.19.0.3:48122: write: broken pipe
2025-02-24T11:44:02.0449152Z   proxy | 2025/02/24 11:44:02 [244] GET https://registry.hub.docker.com:443/v2/hexpm/elixir/tags/list
2025-02-24T11:44:02.0451276Z 2025/02/24 11:44:02 [244] 401 https://registry.hub.docker.com:443/v2/hexpm/elixir/tags/list
2025-02-24T11:44:02.0790947Z   proxy | 2025/02/24 11:44:02 [246] GET https://auth.docker.io:443/token?service=registry.docker.io&scope=repository%3Ahexpm%2Felixir%3Apull&account
2025-02-24T11:44:02.0799472Z   proxy | 2025/02/24 11:44:02 [246] 200 https://auth.docker.io:443/token?service=registry.docker.io&scope=repository%3Ahexpm%2Felixir%3Apull&account
2025-02-24T11:44:02.1109714Z   proxy | 2025/02/24 11:44:02 [248] GET https://registry.hub.docker.com:443/v2/hexpm/elixir/tags/list
2025-02-24T11:44:02.1112752Z 2025/02/24 11:44:02 [248] 200 https://registry.hub.docker.com:443/v2/hexpm/elixir/tags/list
2025-02-24T11:44:02.1122214Z updater | 2025/02/24 11:44:02 INFO <job_969523985> Adding dependencies as handled: (hexpm/elixir).
2025-02-24T11:44:02.1579650Z   proxy | 2025/02/24 11:44:02 [250] POST /update_jobs/969523985/record_update_job_error
2025-02-24T11:44:02.2435527Z   proxy | 2025/02/24 11:44:02 [250] 204 /update_jobs/969523985/record_update_job_error
2025-02-24T11:44:02.2448795Z updater | 2025/02/24 11:44:02 INFO <job_969523985> Handled error whilst updating hexpm/elixir: dependency_file_not_resolvable {:message=>"Error while accessing docker image at registry.hub.docker.com"}
```
</details>

**Comment by robaiken** [comment]
@gmile This seems to happen because `hexpm/elixir` have thousands of tags. The default timeout was 5 seconds when obtaining these tags, I have updated this to 60 seconds  https://github.com/dependabot/dependabot-core/pull/11676

**Comment by gmile** [comment]
Awesome, thank you so much! I'll wait until tomorrow morning to verify if this worked, since it doesn't seem like re-triggering dependabot workflow is possible in principle (attempting to do so returns a warning message in the workflow run)

**Comment by robaiken** [comment]
@gmile To manually retrigger a Dependabot job (it's hidden pretty deep in GitHub's UI):

1. Go to your repository
1. Click the 'Insights' tab at the top
1. Select 'Dependency graph' from the left sidebar
1. Click the 'Dependabot' tab
1. Find your specific dependency alert
1. Click 'Recent update jobs'
1. Press the 'Check for updates' button

This path isn't obvious, it is really buried! Hopefully, this helps

**Comment by wjam** [comment]
I've managed to find an image which isn't getting updated. I've got a private repository which contains the `quay.io/minio/minio:RELEASE.2025-01-20T14-49-07Z` image but dependabot isn't suggesting `RELEASE.2025-02-18T16-25-55Z` which is the current alias to `latest`.

There is very little that I can see in the logs aside from these three lines:

```
2025-02-25T19:39:31.6710531Z updater | 2025/02/25 19:39:31 INFO <job_970622557> Checking if minio/minio RELEASE.2025-01-20T14-49-07Z needs updating
2025-02-25T19:39:31.6714026Z updater | 2025/02/25 19:39:31 INFO <job_970622557> Latest version is RELEASE.2025-01-20T14-49-07Z
2025-02-25T19:39:31.6715237Z 2025/02/25 19:39:31 INFO <job_970622557> No update needed for minio/minio RELEASE.2025-01-20T14-49-07Z
```

**Comment by abdulapopoola** [comment]
@wjam , can you please file a new bug?

**Comment by wjam** [comment]
> [@wjam](https://github.com/wjam) , can you please file a new bug?

Raised #11680 for the issue

**Comment by wjam** [comment]
Also, the GitHub issue labelling currently doesn't understand `docker-compose`!

**Comment by Marinofull** [comment]
Dependabot fails to open for private repositories (in this case, also inside the same organization). Is there a way to make the authentication work or my only option is to skip them?

![Image](https://github.com/user-attachments/assets/4ee5a7ab-6751-47d2-b7f4-03b600235992)

**Comment by robaiken** [comment]
Hi @Marinofull, I see this issue has already been reported here: https://github.com/dependabot/dependabot-core/issues/11686
Would you mind adding your specific details to that thread? Particularly helpful would be:

1. The exact package and version you're attempting to update
2. Your run logs ([instructions on obtaining these logs can be found here](https://docs.github.com/en/code-security/dependabot/troubleshooting-dependabot/viewing-dependabot-job-logs))

Contributing this information helps us better understand the scale of the issue and identify patterns across different environments.

**Comment by Marinofull** [comment]
Thank you @robaiken for your message. It happened that in my case it was only needed to [ignore](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/dependabot-options-reference#ignore--) this private package. The other packages that are public it was able to update as it should, so thank you for the feature!
The issue you mentioned is not related, there it fails for public packages.
